pshtt/pshtt.py

```diff
@@ -1134,30 +1134,46 @@ def load_suffix_list():
     return suffixes
 
 
-def initialize_external_data():
+def initialize_external_data(
+    init_preload_list=None,
+    init_preload_pending=None,
+    init_suffix_list=None
+):
     """
-    This function serves to load all of the third party external data used
+    This function serves to load all of third party external data.
 
     This is meant to be called explicitly by a user. Either the `pshtt` tool
     itself as part of `inspect_domains()` function, or if in a library, as part
     of the setup needed before using certain library functions.
 
-    All downloaded third party data will be cached in a directory, and
-    used from cache on the next pshtt run instead of hitting the network,
-    if the --cache-third-parties=[DIR] flag specifies a directory.
+    If values are passed in to this function, they will be assigned to
+    be the cached values. This allows a caller of the Python API to manage
+    cached data in a customized way.
+
+    It also potentially allows clients to pass in subsets of these lists,
+    for testing or novel performance reasons.
+
+    Otherwise, if the --cache-third-parties=[DIR] flag specifies a directory,
+    all downloaded third party data will be cached in a directory, and
+    used from cache on the next pshtt run instead of hitting the network.
+
+    If no values are passed in, and no --cache-third-parties flag is used,
```

Comment:
@konklone -- Should the caching / interaction with this flag take place in this function? Just to keep the logic together?

Reply:
Yeah, I agree -- and I also think the cache interactions inside the `load_*` functions should move in there as well. The `load_*` functions can become simple functions that only do one thing, and the flag-processing code can just process the flag and get the third party cache dir stored for use in `initialize_external_data()`. I'll push an update to this PR moving that around.